EmployeeManagementSystem/Program.cs

```diff
@@ -0,0 +1,234 @@
+
+        _console.LogInfo("Mobile Number: ");
+        string mobileNumber = (Console.ReadLine() ?? "0");
+
+        _console.LogInfo("Joining Date (MM-DD-YYYY): ");
+        DateTime joiningDate = DateTime.Parse(Console.ReadLine() ?? "0000-00-00");
+
+        _console.LogInfo("Location: (Hyderabad, US, UK)");
+        Location Location;
+        Enum.TryParse(Console.ReadLine(), ignoreCase: true, out Location);
+        
+        _console.LogInfo("Department: (PE, IT)");
+        Department Department;
+        Enum.TryParse(Console.ReadLine(), ignoreCase: true, out Department);
+        
+        _console.LogInfo("Role: (Intern, Developer, Admin)");
+        Role Role;
+        Enum.TryParse(Console.ReadLine(), ignoreCase: true, out Role);
+        
+        _console.LogInfo("Manager: (Hasnu, Sandeep, Bhagvan)");
+        Manager Manager;
+        Enum.TryParse(Console.ReadLine(), ignoreCase: true, out Manager);
+        
+        _console.LogInfo("Project: (p1, p2)");
+        Project Project;
+        Enum.TryParse(Console.ReadLine(), ignoreCase: true, out Project);
+        
+        Employee employee = new Employee
+        {
+            EmpNo = empNo,
+            FirstName = firstName,
+            LastName = lastName,
+            Dob = dob,
+            Mail = mail,
+            MobileNumber = mobileNumber,
+            JoiningDate = joiningDate,
+            Location = Location,
+            Department = Department,
+            Role = Role,
+            Manager = Manager,
+            Project = Project
+        };
+        return employee;
+    }
+    
+    public void Display(List<Employee> employees)
+    {
+        if (employees.Count == 0)
+        {
+            _console.LogError("No employees found");
+            return;
+        }
+        foreach (var employee in employees)
+        {
+            Console.WriteLine(employee.ToString());
+        }
+    } 
+
+    public void FilterAndDisplay()
+    {
+        _console.LogInfo("Enter Alphabet");
+        string alphabetFilter = Console.ReadLine();
+    
+        _console.LogInfo("Enter Location");
+        string locationFilter = Console.ReadLine();
+    
+        _console.LogInfo("Enter Department");
+        string departmentFilter = Console.ReadLine();
+    
+        _console.LogInfo("Enter EmpNo to search");
+        string empNoFilter = Console.ReadLine();
+
+        List<string> filters = new List<string>();
+        if (!string.IsNullOrEmpty(alphabetFilter))
+            filters.Add(alphabetFilter.ToLower()); 
+        if (!string.IsNullOrEmpty(locationFilter))
+            filters.Add(locationFilter.ToLower());
+        if (!string.IsNullOrEmpty(departmentFilter))
+            filters.Add(departmentFilter.ToLower());
+        if (!string.IsNullOrEmpty(empNoFilter))
+            filters.Add(empNoFilter.ToLower());
+
+        List<Employee> filteredEmployees = _bal.Filter(filters);
+        Display(filteredEmployees);
+    }  
+
+    public Employee Update()
```

Comment:
Why return type is is Employee for this? This is just a handler function that is called on selecting option right.